src/openms/include/OpenMS/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.h

```diff
@@ -0,0 +1,273 @@
+    from a DDA experiment) based on a target list.
+
+    The input experiment is expected to be a standard .mzML file.
+    The input target list is expected to be a standard TraML formatted .csv file.
+
+    The class deals with filtering and annotating only those spectra that reflect
+    the characteristics described by the target list.
+    The filtering is done based on the transition_name, PrecursorMz and RetentionTime.
+    The spectra are smoothed and peaks are picked for each spectrum.
+    The spectra are then scored based on total TIC, FWHM and SNR.
+    One spectrum is chosen for each of those transitions for which at least one
+    valid spectrum was found and matched.
+  */
+  class OPENMS_DLLAPI TargetedSpectraExtractor :
+    public DefaultParamHandler
+  {
+public:
+    TargetedSpectraExtractor();
+    virtual ~TargetedSpectraExtractor();
+
+    void setRTWindow(const double& rt_window);
+    double getRTWindow() const;
+
+    void setMinScore(const double& min_score);
+    double getMinScore() const;
+
+    void setMinForwardMatch(const double& min_forward_match);
+    double getMinForwardMatch() const;
+
+    void setMinReverseMatch(const double& min_reverse_match);
+    double getMinReverseMatch() const;
+
+    void setMZTolerance(const double& mz_tolerance);
+    double getMZTolerance() const;
+
+    void setMZUnit(const bool& mz_unit_is_Da);
+    bool getMZUnit() const;
+
+    void setSGolayFrameLength(const UInt& sgolay_frame_length);
+    UInt getSGolayFrameLength() const;
+
+    void setSGolayPolynomialOrder(const UInt& sgolay_polynomial_order);
+    UInt getSGolayPolynomialOrder() const;
+
+    void setGaussWidth(const double& gauss_width);
+    double getGaussWidth() const;
+
+    void setUseGauss(const bool& use_gauss);
+    bool getUseGauss() const;
+
+    void setSignalToNoise(const double& signal_to_noise);
+    double getSignalToNoise() const;
+
+    void setPeakHeightMin(const double& peak_height_min);
+    double getPeakHeightMin() const;
+
+    void setPeakHeightMax(const double& peak_height_max);
+    double getPeakHeightMax() const;
+
+    void setFWHMThreshold(const double& fwhm_threshold);
+    double getFWHMThreshold() const;
+
+    void setTICWeight(const double& tic_weight);
+    double getTICWeight() const;
+
+    void setFWHMWeight(const double& fwhm_weight);
+    double getFWHMWeight() const;
+
+    void setSNRWeight(const double& snr_weight);
+    double getSNRWeight() const;
+
+    void getDefaultParameters(Param& params);
+
+    /**
+      @brief Filters and annotates those spectra that could potentially match the
+      transitions of the target list.
+
+      The spectra taken into account are those that fall within the precursor RT
+      window and MZ tolerance set by the user through the setRTWindows() and
+      setMZTolerance setters. Default values are provided for both parameters.
+
+      @warning The picked spectrum could be empty, meaning no peaks were found.
```

Comment:
what happens if two spectra match within the m/z and RT tol?

Reply:
they will match the same transition, and then `selectSpectra` would select the one with the highest score for that transition